### PR TITLE
Add formatting buttons to the WysiwygComposer

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Notes: Add formatting buttons to the rich text editor

- New formatting buttons which reflect the internal state of the wysiwyg editor
- The typescript definition of the `matrix-wysiwyg` is still febrile and it will be improve in the future. That's why we need to do the `ReturnType`.  
- The `useWysiwyg` hook for `matrix-wysiwyg` doesn't have a `onChange` handler anymore but returns directly the `content` alongside of the `formattingStates`.
- The formatting buttons increase the height of the editor and emoji, attachment... buttons weren't aligned. I added an ugly workaround CSS to fix it. In one of the next PR, the wysiwyg visual will be close to the expected visual and it will need to reorganise the bottom part of the room. For example, the emoji button and the encrypted icon will be outlined with the editor. It'll change the component structure and the related CSS. That's why a temporary workaround is used here.
- The Wysiwyg lab flag is renamed to `Try out the rich text editor (plain text mode coming soon)`
- Also, I would like to not mock the `useWysiwyg` hook and make it works in the test environment. Currently, we are mocking the fetch in `matrix-wysiwyg` repository (https://github.com/matrix-org/matrix-wysiwyg/blob/main/platforms/web/test.setup.ts) to return the wasm file. Here the wasm is in the js file... With the deadline, sadly I don't have extra time to make it work here....

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add formatting buttons to the rich text editor ([\#9410](https://github.com/matrix-org/matrix-react-sdk/pull/9410)). Contributed by @florianduros.

### CODE REVIEW

File: src/components/views/rooms/wysiwyg_composer/WysiwygComposer.tsx [hunk 1]
```diff
@@ -53,18 +57,8 @@ export function WysiwygComposer(
...
-                    contentEditable={!disabled && isWysiwygReady}
-                    role="textbox"
-                    aria-multiline="true"
-                    aria-autocomplete="list"
-                    aria-haspopup="listbox"
-                    dir="auto"
-                    aria-disabled={disabled || !isWysiwygReady}
-                />
-            </div>
+            <FormattingButtons wysiwyg={wysiwyg} formattingStates={formattingStates} />
```
Comment (gsouquet) [inline src/components/views/rooms/wysiwyg_composer/WysiwygComposer.tsx:60]: Probably a minor nitpick, and feel free to punt that comment to a later stage, but rather than naming props `wysiwyg` I think we should call them `composer`.

It's only a temporary thing that we have two different composer in the codebase, the ultimate goal would be to have this one taking over, and when that happens we will end up having a `wysiwyg` prop that's hard to parse.

What do you think?

File: src/components/views/rooms/wysiwyg_composer/FormattingButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        title={label}
+        className={
+            classNames('mx_FormattingButtons_Button', className, { 'mx_FormattingButtons_active': isActive })}
+        tooltip={keyCombo && <Tooltip label={label} keyCombo={keyCombo} />}
+        alignment={Alignment.Top}
+    />;
+}
+
+interface FormattingButtonsProps {
+    wysiwyg: ReturnType<typeof useWysiwyg>['wysiwyg'];
```
Comment (gsouquet) [inline src/components/views/rooms/wysiwyg_composer/FormattingButtons.tsx:58]: Same comment about naming applies here

**Review by gsouquet — APPROVED** [review]
Looks good to me!

File: src/components/views/rooms/wysiwyg_composer/WysiwygComposer.tsx (hunk 1 above)
Comment (florianduros) [reply src/components/views/rooms/wysiwyg_composer/WysiwygComposer.tsx:60]: I agree with your idea, I'll rename it!

File: src/components/views/rooms/wysiwyg_composer/FormattingButtons.tsx (hunk 2 above)
Comment (florianduros) [reply src/components/views/rooms/wysiwyg_composer/FormattingButtons.tsx:58]: I agree with your idea, I'll rename it!
